libs/platform/experience/src/services/layout/default-layout.service.ts

```diff
@@ -1,15 +1,59 @@
 import { ssrAwaiter } from '@spryker-oryx/core/utilities';
 import { inject } from '@spryker-oryx/di';
-import { Breakpoint, sizes } from '@spryker-oryx/utilities';
-import { merge, Observable, of } from 'rxjs';
-import { reduce } from 'rxjs/operators';
+import { Breakpoint, Size, sizes, throttle } from '@spryker-oryx/utilities';
+import {
+  merge,
+  Observable,
+  of,
+  ReplaySubject,
+  distinctUntilChanged,
+  map,
+  reduce,
+} from 'rxjs';
 import { CompositionLayout } from '../../models';
 import { LayoutStyles, ResponsiveLayoutInfo } from './layout.model';
 import { LayoutService } from './layout.service';
 import { ScreenService } from './screen.service';
 
 export class DefaultLayoutService implements LayoutService {
-  constructor(protected screenService = inject(ScreenService)) {}
+  constructor(protected screenService = inject(ScreenService)) {
+    this.setupBreakpointsObserver(document.body);
+  }
+
+  protected screenWidth$ = new ReplaySubject<number>();
+
+  protected setupBreakpointsObserver(target: HTMLElement): void {
+    if (!target) return;
+
+    this.screenWidth$.next(window.innerWidth);
+
+    const observer = new ResizeObserver(
+      throttle(
+        () =>
+          window.requestAnimationFrame(() => {
+            this.screenWidth$.next(window.innerWidth);
+          }),
+        50
+      )
+    );
+    observer.observe(target);
+  }
+
+  protected evaluateBreakpoint(width: number): Size {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return Object.entries(this.screenService.getBreakpoints()).find(
+      ([_b, { min = 0, max = Infinity }]) => {
+        return width >= min && width <= max;
+      }
+    )![0] as Size;
+  }
+
+  getBreakpoint(): Observable<Size> {
+    return this.screenWidth$?.pipe(
+      map((width) => this.evaluateBreakpoint(width)),
+      distinctUntilChanged()
+    );
+  }
```

Comment:
Looks like these functionality closer to `ScreenService` then to layout

Reply:
Not sure. Screen service is like a mediator between app and theme. I agree that method's name could be not very convenient, renamed it to `getActiveBreakpoint`

Reply:
Discussed with @tobi-or-not-tobi and moved related logic to the screen service